### ISSUE TITLE
Loss function

### ISSUE THREAD

**Issue by JeonMinkyu** [grey-eye/talking-heads#13]
In loss.py, 
class LossD(nn.Module):
    def __init__(self, device):
        super(LossD, self).__init__()
        self.dtype = torch.FloatTensor if device == 'cpu' else torch.cuda.FloatTensor
        self.zero = torch.zeros([1]).type(self.dtype)

    def forward(self, r_x, r_x_hat):
        #r_x : D(x_t,y_t,i)'s result
        #r_x_hat : D(x_hat, y_t, i)'s result
        return (1 + r_x_hat) + (1 - r_x) 
-> In paper, LossD is hingeLoss (L_DSC) , but I think it's not hinge loss.

Did I miss something?

**Comment by subrtade662** [comment]
I agree, I've tried to use MarginRankingLoss but the discriminator is still too weak.

**Comment by MrCaracara** [comment]
You're correct, I am not using hinge loss. That's simply because I never got around to fixing this. PyTorch doesn't provide a hinge loss function besides `torch.nn.HingeEmbeddingLoss`, which doesn't fit this use case, and I had some trouble using the max function for tensors. In order to get the script running I simply removed the max. But this is not the way the loss function is supposed to look!

**Comment by JeonMinkyu** [comment]
I used Hingeloss, and verified(identified) the loss whether it is appropriate in my task. If you dont mind, Could you leave the comment for my task. It is confusing to me if I use D loss value, and whether it is changed with negative value or not.

**Comment by MrCaracara** [comment]
Yeah it's fine. I saw your pull request and I will merge it.
